.claude/hooks/generated-files-guard.sh

```
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+# Get the project root directory
+PROJECT_ROOT="$(cd "$(dirname "$0")/../.." && pwd)"
+HOOK_DIR="$PROJECT_ROOT/.claude/hooks"
+
+# Change to project root
+cd "$PROJECT_ROOT"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)